src/dbwrapper.cpp

```diff
@@ -100,7 +100,7 @@ class CBitcoinLevelDBLogger : public leveldb::Logger {
 
                 assert(p <= limit);
                 base[std::min(bufsize - 1, (int)(p - base))] = '\0';
-                LogPrintLevel(BCLog::LEVELDB, BCLog::Level::Debug, "%s", base); // NOLINT(bitcoin-unterminated-logprintf)
+                LogPrintLevel(LogInstance(), BCLog::LEVELDB, BCLog::Level::Debug, "%s", base); // NOLINT(bitcoin-unterminated-logprintf)
```

Comment:
This makes the call sites even more verbose and `LogInstance(), ` is repeated everywhere. In this case one improvement could be to use a default argument for that parameter which defaults to `LogInstance()` but I can see that this may not play well with the `...` arguments.

Maybe introduce another function that takes `(category, level, instance, ...)`? Or check if the first of `...` arguments in `(category, level, ...)` is a log instance and if not, then default the instance to `LogInstance()`? Something like:

```cpp
  class Logger
  {
  };
  
  template <typename... Args>
  void Log(int, const Logger&, const Args&...)
  {
  }
  
  template <typename... Args>
  void f(int a, const Args&... args)
  {
      auto CallLog = [&](const auto& first, const auto&... rest) {
          if constexpr (std::is_same<decltype(first), Logger>::value) {
              Log(a, first, rest...);
          } else {
              Log(a, Logger{}, first, rest...);
          }
      };
  
      CallLog(args...);
  }

  int main(int, char**)
  {
      f(5, 6);
      f(8, Logger{}, 9);
      return 0;
  }
```

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30338#discussion_r1654817299

> This makes the call sites even more verbose and `LogInstance()`, is repeated everywhere

See #30342 for an alternative which should make everything less verbose. For example, this line changes to `LogDebug(m_log, "%s", base);`